### PR TITLE
Add optional timeout to lock

### PR DESCRIPTION
**Pull request by tsholmes**
This will allow integrations to optionally specify how long to lock for (specifically for the case where they have no intention of unlocking).

This will allow us to convert marketo to a uniform locking method, and handle failed locks with a backoff retry, instead of hard failing the message after a few futile (internal) attempts.

### CODE REVIEW

**Comment by tsholmes** [comment]
cc @ndhoule @f2prateek

**Comment by f2prateek** [comment]
LGTM - this is mainly so that marketo can use the standard locking instead of the custom one in marketo-  https://github.com/segment-integrations/integration-marketo/blob/master/lib/marketo/lock.js ?

**Comment by tsholmes** [comment]
Yeah. That one locks for 30s, but only retries internally 3 times with a second between attempts before returning an error that results in it not being retried by the normal mechanisms.

File: lib/proto.js [hunk 1]
```diff
@@ -183,10 +183,14 @@ exports.map = function(events, str){
  * @api public
  */
 
-exports.lock = function(key, fn){
+exports.lock = function(key, timeout, fn){
+  if (fn === undefined) {
```
Comment (ndhoule) [inline lib/proto.js]: You can just check `!fn` here

**Pull request merged**